src/chapters/classes-part2/equals-shortcut.rst

```diff
@@ -0,0 +1,72 @@
+Visual Studio Generator Shortcut
+================================
+
+Instead of cutting, pasting, and refactoring old code to ensure that you create a well-structured ``GetHashCode()`` method whenever you define your own ``Equals()`` method, you can use Visual Studio’s code generation tool!
+Just right-click in your class file on the class name and follow these prompts.
+
+Let’s use a Course class to demonstrate:
+
+.. sourcecode:: csharp
+   :linenos:
+
+   public class Course
+    {
+            public string Topic { get; set; }
+            public Teacher Instructor { get; set; }
+            public List<Student> enrolledStudents { get; set; }
+    }
+
+#. In Visual Studio, right-click on the class name and select *Quick Fix* from the menu.
+
+   .. figure:: figures/select-quick-fix.png
+      :alt: Screenshot showing the Quick Fix option at the top of the menu when one right-clicks on the class name.
+
+#. Select the *Generate Equals() and GetHashCode()* option from the resulting menu:
+
+   .. figure:: figures/generate-equals.png
+      :alt: Screenshot showing the Generate Equals() and GetHashCode() option in the pop up.
+
+#. A new menu will appear where you can make selections about which class members you want to use for the ``Equals()`` and ``GetHashCode()`` methods.
+   In the case of equality between two ``Course`` objects, we want to establish that two ``Course`` objects are equal if the ``Topic`` members and ``Instructor`` members are equal.
+   Once you select the necessary members, click *Ok*!
+
+   .. figure:: figures/make-selections.png
+      :alt: Screenshot showing the topic and instructor members selected in a pop-up.
+
+#. Once you click *Ok*, Visual Studio generates the ``Equals()`` and ``GetHashCode()`` methods, resulting in the following class.
+
+   .. sourcecode:: csharp
+      :linenos:
+
+      public class Course
+      {
+            public string Topic { get; set; }
+            public Teacher Instructor { get; set; }
+            public List<Student> enrolledStudents { get; set; }
+
+            public override bool Equals(object obj)
+            {
+                return obj is Course course &&
+                   Topic == course.Topic &&
+                   EqualityComparer<Teacher>.Default.Equals(Instructor, course.Instructor);
+            }
+
+            public override int GetHashCode()
+            {
+                return HashCode.Combine(Topic, Instructor);
+            }
+      }
+
+In order to gain an understanding at what Visual Studio just did for us, review the section on the :ref:`Equals() method <equals-method>` and take note of the following lines of code in the code block above.
+Visual Studio's method does not necessarily look similar to the ones we wrote previously.
```

Comment:
what about adding something about how even though this syntax is different, the comparisons are basically the same?
also, if this Equals method can use lowercase object, can we also use it for the example equals on the previous page?